### PR TITLE
[Java 21] Remove Gson library usage from the Profiler

### PR DESCRIPTION
**Pull request by ravinperera00**
## Check List 
- [X] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-profiler/build.gradle [hunk 1]
```diff
@@ -25,7 +25,6 @@ configurations {
 dependencies {
     implementation "org.ow2.asm:asm:${project.ow2AsmVersion}"
     implementation "org.ow2.asm:asm-commons:${project.ow2AsmCommonsVersion}"
-    implementation "com.google.code.gson:gson:${project.gsonVersion}"
```
Comment (warunalakshitha) [inline bvm/ballerina-profiler/build.gradle:28]: Shall we remove from rt jar as well

Comment (ravinperera00) [reply bvm/ballerina-profiler/build.gradle:28]: It is needed for observability right?

Comment (ravinperera00) [reply bvm/ballerina-profiler/build.gradle:28]: Resolved through external discussion.

(Fixed in f7f7973d0a9481e6e3070cda49b1baac2efbef5c)

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**